freva-rest/src/databrowser_api/core.py

```diff
@@ -1127,356 +1122,175 @@ def escape_special_chars(value: str) -> str:
-        else:
-            time_str = "fx"
-
-        dt = abs((times[1] - times[0]).total_seconds()) if len(times) > 1 else 0
-
-        variables = [
-            var for var in data_vars
-            if var not in coords
-            and not any(term in var.lower() for term in
-                        ["lon", "lat", "bnds", "x", "y"])
-            and var.lower() not in ["rotated_pole", "rot_pole"]
-        ]
-
-        if len(variables) != 1:
-            logger.error("Only one data variable allowed, found: %s", variables)
-
-        _data = self.fwrites.copy()
-
-        _data.setdefault("variable", variables[0])
-        _data.setdefault("time_frequency", self.get_time_frequency(dt, time_freq))
-        _data["time"] = time_str
-        _data.setdefault("cmor_table", _data["time_frequency"])
-        _data.setdefault("version", "")
-        _data["file"] = str(file_name)
-        _data["uri"] = str(file_name)
-
-        return cast(Dict[str, Union[str, list[str], Dict[str, str]]], _data)
-
-    async def add_userdata(
-        self, args: Tuple[str, list[os.PathLike[str]], Mapping[str, str]]
+    async def add_user_metadata(
+        self, user_name: str, user_metadata: List[Dict[str, Any]],
+        **fwrites: Dict[str, str]
     ) -> None:
         """
-        Add user data to the Apache Solr search system and MongoDB.
-
-        This method validates user data paths, prepares the necessary information,
-        and ingests the data into the Apache Solr and MongoDB systems.
-
-        Parameters:
-        ~~~~~~~~~~
-        - user (str): The identifier of the user whose data is being added.
-        - paths (List[os.PathLike[str]]): validated user data paths.
-        - **fwrites (Dict[str, str]): Key-value pairs representing additional
-          metadata to be written,
+        Add validated user metadata to the Apache Solr search system
+        and MongoDB.
 
-        Returns:
-        ~~~~~~~~
-        - None
+        parameters:
+        ----------
+        user_name: str
+            The username associated with the metadata.
+        user_metadata: List[Dict[str, Any]]
+            The metadata to be ingested.
+        fwrites: Dict[str, str]
+            Optional additional file write parameters.
         """
-        self.executor = concurrent.futures.ThreadPoolExecutor(
-            max_workers=min(mp.cpu_count(), 15)
-        )
-
+        max_workers = min(mp.cpu_count(), 15)
+        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=max_workers)
+        self.fwrites |= {
+            "user": user_name,
+            "fs_type": self.fs_type
+        } | fwrites.get("facets", {})
         try:
-            user, paths, fwrites = args
-            self.validated_userdata = paths
-            self.fwrites |= {"user": user,
-                             "fs_type": self.fs_type} | cast(Dict[str, str], fwrites)
-            await self._ingest_user_data()
-
+            self.validated_user_metadata = user_metadata
```

Comment:
I think you want to pass user_metdata as an argument rather than assigning it to a class attribute. The submit function takes the metadata list as an argument anyway. So you might just pass it through.